test/inductor/test_compiled_optimizers.py

```diff
@@ -179,24 +178,60 @@ def fn():
         def fn():
             step_fn(opt_compiled)
 
-    return torch.compile(fn, backend="inductor", fullgraph=True)
+    return torch.compile(fn, backend="inductor", fullgraph=fullgraph)
+
+
+def check_optim(
+    test_cls, optim_cls, params_eager, params_compiled, state_eager, state_compiled
+):
+    params_eager = list(params_eager)
+    params_compiled = list(params_compiled)
+    # Note on tolerances:
+    # test_correctness_Adadelta_cuda_float32
```

Comment:
This should live in common_optimizers.py, something like https://github.com/pytorch/pytorch/blob/b18d8d4595aa6e0768eedd5fc7d4a4402c567181/torch/testing/_internal/common_optimizers.py#L1087-L1094

Reply:
I think this will have to wait for a larger refactor, because this check function is used by both the kernel count tests and the correctness tests.

Reply:
Let's add these for the new `test_correctness` ones anyway to consolidate there.

Reply:
Yeah this won't work because I generate the tests differently for the compiled optimizers (it won't take into account the DecorateInfo. The reason for that separation is that we generate a test per config for easier debugging.

Reply:
I think we commented at the same time, I can add them there as well